15-greenhouse-monitor/README.md

```diff
@@ -0,0 +1,369 @@
+
+#### Using the In-browser terminal
+
+1. Connect the Notecarrier to your computer using a micro-USB cable.
+2. Launch the in-browser terminal at [blues.dev](https://dev.blues.io/notecard-playground/)
+3. Click the "USB Notecard" button under "Connect a Notecard".
+4. Select the Notecard to connect to and click "Connect".
+5. The terminal will display the firmware version of Notecard.
+6. You can now enter a request to set the ProductUID and Serial Number of the device.
+
+```json
+{"req":"hub.set", "product":"<your-productUID-from-notehub>", "sn":"greenhouse-monitor"}
+```
+
+You can also omit the serial number use Notehub to set it:
+
+1. Open the project in Notehub.
+2. From the list of devices, double click the device whose serial number you want to set.
+3. In the "Summary" tab, use the pencil icon to edit the "Serial Number" field.
+
+#### Editing the Source Code
+
+You can also set the `ProductUID` in the source code. Open `app.h` in your IDE and edit the line
+
+```c++
+#define PRODUCT_UID ""		// "com.my-company.my-name:my-project"
+```
+
+pasting in the `ProductUID` from your Notehub project between the first pair of quotes.
+
+
+## Testing
+
+To ensure the setup is working as expected, it's a good idea to test the application before deploying it in a real-life setting. The following sections describe how to configure and use the app.
+
+### App Configuration
+
+The app is configured using environment variables to set
+
+* how often environment variables are checked,
+* how often the sensors are read and checked for alerts, and
+* how often monitoring events are sent.
+
+These variables control the overall app behavior:
+
+* `environment_update_mins` How often (in minutes) to check for environment variable updates from the Notecard. When not set, the default value is 5 minutes.
+
+* `monitor_secs`: How often (in seconds) to read the current values from the sensors and check for alerts. When not set, the default value is 15 seconds.
+
+* `report_mins` How often (in minutes) to report the current values from sensors to `greenhouse.qo`. When not set, the default value is 5 minutes.
+
+You set these environment variables in Notehub. See our tutorial [Understanding Environment Variables](https://dev.blues.io/guides-and-tutorials/notecard-guides/understanding-environment-variables/) for a fuller description of how to set environment variables in Notehub.
+
+
+### Sensor Names
+
+The app senses and reports these values:
+
+* `air_temp`: the air temperature in Celsius
+
+* `air_humidity`: the relative humidity as a percent
+
+* `air_pressure`: the air pressure in kPa
+
+* `soil_moisture`: an indication of how much moisture is in the soil. A higher value means more moisture, a lower value means less moisture.
```

Comment:
This is kind of vague. If that's just a limitation of the sensor, though, that makes sense. But knowing the scale at least would help, I think, even if we can't know the unit of measurement. Same goes for `light_level`.

Reply:
I now see you go over this in greater detail in the "Soil Moisture and Light Level" section. Maybe just adding a link to that part of the document here would be good, in that case.